glean-core/android/src/main/java/mozilla/telemetry/glean/testing/GleanTestLocalServer.kt

```diff
@@ -31,12 +35,23 @@ import org.junit.runner.Description
  */
 @VisibleForTesting(otherwise = VisibleForTesting.NONE)
 class GleanTestLocalServer(
+    val context: Context,
     private val localPort: Int
 ) : TestWatcher() {
     /**
      * Invoked when a test is about to start.
      */
     override fun starting(description: Description?) {
         Glean.testSetLocalEndpoint(localPort)
+
+        val config = Configuration.Builder()
+            // Use a single thread executor, we cannot make background upload
```

Comment:
I'm not sure I understand how a single-thread executor means its not on the main thread?
Is this "single thread" separate from where it is started?

Reply:
Good point. I tried to rephrase the comment a bit to clarify. The problem is that the default test executor from the WorkManager runs things on the main thread, and we don't want that, as networking would crash the test. Instead we just use the simplest possible executor: a pool with 1 thread.